app/Http/Controllers/CategoryController.php

```diff
@@ -49,12 +39,12 @@ public function update(CategoryRequest $request, Category $category)
 
     public function destroy(string $categoryId)
     {
-         $category = Category::find($categoryId);
+        $category = Category::where('id', $categoryId)->first();
 
         if ($category->children->isNotEmpty()) {
```

Comment:
tamare te check karvanu che k $categoryId parent_id ma che k nahi. jo hoy to means te koi child no parent che. 
`if ($category->children->isNotEmpty()) {` not required